session/git/worktree_pattern.go

```diff
@@ -0,0 +1,113 @@
+package git
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+)
+
+// PatternVariables holds the available variables for worktree pattern substitution
+type PatternVariables struct {
+	RepoRoot    string
+	RepoName    string
+	IssueNumber string
+	Title       string
+	Timestamp   string
+}
+
+// parseWorktreePattern substitutes variables in the pattern with actual values
+func parseWorktreePattern(pattern string, vars PatternVariables) string {
```

Comment:
I think you can use`text/template` to do this replacing operation.

Reply:
Ah I will update to this, pretty new to go so still learning alot of things.

Reply:
Updated via 5908ac6c42fd4f7dc6f247e41eb091228d33b83e